src/integration-tests/bash/cleanup.sh

```diff
@@ -523,32 +523,38 @@ if [ -x "$(command -v helm)" ]; then
   [[ $? == 1 ]] && HELM_VERSION=V3
   echo "Detected Helm Version [$(helm version --short --client)]"
   echo @@ `timestamp` Deleting installed helm charts
-  namespaces=`kubectl get ns | grep -v NAME | awk '{ print $1 }'`
-  for ns in $namespaces
-  do 
+  if [ "$HELM_VERSION" == "V2" ]; then
+   helm list --short | while read helm_name; do
    if [ ! "$DRY_RUN" = "true" ]; then
-     (
+   (
      set -x
-     helm list --short --namespace $ns | while read helm_name; do
-       if [ "$HELM_VERSION" == "V2" ]; then
-         helm delete --purge  $helm_name
-       else 
-         helm uninstall $helm_name -n $ns 
-       fi
-     done
-     )
+     helm delete --purge  $helm_name
+    )
    else
-     (
-     helm list --short --namespace $ns | while read helm_name; do
-       if [ "$HELM_VERSION" == "V2" ]; then
-         echo @@ `timestamp` Info: DRYRUN: helm delete --purge  $helm_name
+     echo @@ `timestamp` Info: DRYRUN: helm delete --purge  $helm_name
+   fi
+   done
+  fi
+
+  if [ "$HELM_VERSION" == "V3" ]; then
+   namespaces=`helm list --all-namespaces | grep -v NAME | awk '{print $2}'`
```

Comment:
I suspect this should be:      

helm list --all-namespaces | grep -v NAME | awk '{print $2}' | sort -u

We only want to iterate through each namespace once.

Reply:
nice suggestion. will implement change in next commit

Reply:
i am assuming that the above command does not yield a ns unless there's at least one helm release in the ns